### PR TITLE
Build most of Baldr with -Werror

### PR DESCRIPTION
**Pull request by dillona**

### CODE REVIEW

**Comment by kevinkreiser** [comment]
Thank you for the cleanup and happy new year! I'll review this tomorrow but one thing I noticed is that your changelog entry should be at the bottom of the subsection it's in not at the top. And I may have seen some whitespace discrepencies in the cmakelists.txt changes I'll look closer tomorrow

File: src/baldr/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +71,12 @@ set(sources
     verbal_text_formatter_us_tx.cc
     verbal_text_formatter_factory.cc)
 
+set (sources_with_warnings
+    graphreader.cc
+    graphtile.cc
+    graphtileheader.cc
+    )
```
Comment (nilsnolde) [inline src/baldr/CMakeLists.txt:78]: We have the same convention for CMakeLists.txt as for C++ files where the parentheses closes on the same line as the last element

File: src/baldr/CMakeLists.txt [hunk 2]
```diff
@@ -89,12 +92,17 @@ list(APPEND sources
     ${CMAKE_CURRENT_BINARY_DIR}/date_time_southamerica.h
     ${CMAKE_CURRENT_BINARY_DIR}/date_time_systemv.h
     ${CMAKE_CURRENT_BINARY_DIR}/date_time_windows_zones.h
+    )
```
Comment (nilsnolde) [inline src/baldr/CMakeLists.txt:95]: same here

**Review by nilsnolde — CHANGES_REQUESTED** [review]
Jep thanks. As Kevin said, just a few whitespace issues and the readme entry to the bottom.

Happy new year:)

File: src/baldr/merge.cc [hunk 3]
```diff
@@ -140,14 +140,15 @@ std::pair<GraphId, GraphId> edge_collapser::nodes_reachable_from(GraphId node_id
...
 GraphId edge_collapser::next_node_id(GraphId last_node_id, GraphId node_id) {
-  //
-  //        -->--     -->--
-  //   \   /  e4 \   /  e1 \   /
-  //   -(p)       (c)       (n)-
-  //   /   \ e3  /   \ e2  /   \
-  //        --<--     --<--
-  //
-  // given p (last_node_id) and c (node_id), return n if there is such a node.
+  /*
```
Comment (kevinkreiser) [inline src/baldr/merge.cc:143]: can you also revert this change. its best to avoid whitespace-only changes and instead only make changes that fix an issue. unless there is some change that im not detecting here other than the commenting type?

**Review by kevinkreiser — CHANGES_REQUESTED** [review]

File: src/baldr/merge.cc (hunk 3 above)
Comment (dillona) [reply src/baldr/merge.cc:143]: This generated a warning (and therefore error) on Ubuntu 20.04.5 on ARM:

```
/home/ubuntu/valhalla/src/baldr/merge.cc:147:3: error: multi-line comment [-Werror=comment]
  147 |   //   /   \ e3  /   \ e2  /   \
      |   ^
cc1plus: all warnings being treated as errors
make[3]: *** [src/baldr/CMakeFiles/valhalla-baldr.dir/build.make:466: src/baldr/CMakeFiles/valhalla-baldr.dir/merge.cc.o] Error 1
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,6 +22,7 @@
    * CHANGED: Replace unstable c++ geos API with a mix of geos' c api and boost::geometry for admin building [#3683](https://github.com/valhalla/valhalla/pull/3683)
    * ADDED: optional write-access to traffic extract from GraphReader [#3876](https://github.com/valhalla/valhalla/pull/3876)
    * UPDATED: locales from Transifex [#3879](https://github.com/valhalla/valhalla/pull/3879)
+   * FIXED: Build most of Baldr with -Werror [#3885](https://github.com/valhalla/valhalla/pull/3885)
```
Comment (kevinkreiser) [inline CHANGELOG.md:25]: bug fixes go at the bottom of the "bug fix" subsection. i would probably call this a change though not a bug, would you mind changing this to changed?
```suggestion
   * CHANGED: Build most of Baldr with -Werror [#3885](https://github.com/valhalla/valhalla/pull/3885)
```

File: src/baldr/merge.cc (hunk 3 above)
Comment (dillona) [reply src/baldr/merge.cc:143]: With GCC 9.4.0:
```
$ gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

File: CHANGELOG.md (hunk 4 above)
Comment (dillona) [reply CHANGELOG.md]: Yes no problem

File: src/baldr/CMakeLists.txt (hunk 1 above)
Comment (dillona) [reply src/baldr/CMakeLists.txt]: Thank you I have changed this

File: src/baldr/CMakeLists.txt (hunk 2 above)
Comment (dillona) [reply src/baldr/CMakeLists.txt]: Thank you I have changed this

File: src/baldr/merge.cc (hunk 3 above)
Comment (kevinkreiser) [reply src/baldr/merge.cc:143]: oh man multiline comment as an error... that is a new level of pedantic..

Comment (kevinkreiser) [reply src/baldr/merge.cc:143]: ok i guess that comes from -Wall which we didnt have before we just had -Werror. looking at the various namespaces in the project its clear that we havent yet decided what level of  warning is desirable for us, some are just -Werror some add specific ones some add -Wextra as you have here. I think for now we can let it as you have it but at some point we should decide what level or warning should be treated as  error and commit to that across the project. i have a feeling i'll regret this suggestion though as i tend to hate overly perscriptive compilers and linters and pretty much everyone else likes the opposite :smile:

**Review by kevinkreiser — APPROVED** [review]

File: src/baldr/merge.cc (hunk 3 above)
Comment (dillona) [reply src/baldr/merge.cc:143]: Yeah I copied and pasted the CMake infrastructure changes from (I think) Mjolnir. I think the `sources`/`sources_with_warnings` structure is good and I would probably advocate for standardizing on a common structure and arguments at the top level

**Comment by dillona** [comment]
Thanks for the reviews and happy new year @kevinkreiser and @nilsnolde
